src/main/java/com/google/sps/TripDay.java

```diff
@@ -99,7 +105,34 @@ public LocalDate getDate() {
   }
 
   /**
-   * Add locations
+   * Build location entities to be put in datastore. 
+   * @param locations ArrayList of locations (in optimized order) as strings
+   * @param parentKeyId Key of TripDay entity that these locations are a part of  
+   */
+  public static List<Entity> locationsToEntities(List<String> locations, Key parentKeyId) {
+    List<Entity> locationEntities = new ArrayList<>();
+    for (int i = 0; i < locations.size(); i++) {
+      Entity locationEntity = new Entity(LOCATION_ENTITY_TYPE, parentKeyId);
+      locationEntity.setProperty(NAME, locations.get(i));
+      locationEntity.setProperty(ORDER, i);
+      locationEntities.add(locationEntity);
+    }
+    return locationEntities;
+  } 
+
+  /**
+   * Store list location entities to be put in datastore.
+   * @param locationEntities ArrayList of Entities to be put in Datastore
+   * @param datastore Datastore for storing 
+   */
+  public static void storeLocationsInDatastore(List<Entity> locationEntities, DatastoreService datastore) {
+    for (Entity locationEntity : locationEntities) {
+      datastore.put(locationEntity);
+    }
+  }
+
+  /**
+   * Add/set locations
    */
   public void setLocations(ArrayList<String> locations) {
     List<String> temp = new ArrayList<>();
```

Comment:
tip; `new ArrayList<>(locations);` will construct a new arraylist containing all elements from locations in order in a single line.

Reply:
Fixed!